packages/react/src/useActionForm.ts

```diff
@@ -67,6 +67,39 @@ const toDefaultValues = (modelApiIdentifier: string | undefined, data: any) => {
   return data;
 };
 
+const disambiguateDefaultValues = (data: any, initialData: any, action: any) => {
+  const initialKeys = Object.keys(initialData).filter((key) => !OmittedKeys.includes(key as OmittedKey));
+
+  const result = Object.fromEntries(
+    Object.entries(data).flatMap(([key, value]) => {
+      if (initialKeys.includes(key)) {
+        return [];
+      }
+
+      return [[key, value]];
+    })
+  );
+
+  const modelData = { ...data[action.modelApiIdentifier] };
+
+  for (const key of Object.keys(modelData)) {
+    const initialValue = initialData[key];
+
+    if (
+      typeof data[action.modelApiIdentifier] === "object" &&
```

Comment:
`typeof null` is `object`, thanks javascript. this is what lodash does
https://github.com/lodash/lodash/blob/3.0.2-npm-packages/lodash.isobject/index.js

Reply:
:heart: